Http/HttpUtils.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using UnityEngine;
+
+namespace DUCK.Http
+{
+	/// <summary>
+	/// A utility class for http, deals with response codes and error messages.
+	/// <see cref="https://developer.mozilla.org/en-US/docs/Web/HTTP/Status"/>
+	/// </summary>
+	public static class HttpUtils
+	{
+		private static readonly Dictionary<ResponseType, string> responseTypeMessages = new Dictionary<ResponseType, string>
+		{
+			{ResponseType.Unknown, "Cannot find server - {1}"},
+			{ResponseType.Successful, "Request successful with code: {0} - {1}"},
+			{ResponseType.Error, "The client encounted an error with code: {0} - {1}"},
+			{ResponseType.ServerError, "The server encounted an error and responded with code: {0} - {1}"}
+		};
+
+		private static readonly Dictionary<long, string> customResponseTypeMessages = new Dictionary<long, string>();
+
+		/// <summary>
+		/// Format and append parameters to a uri
+		/// </summary>
+		/// <param name="uri">The uri to append the properties to.</param>
+		/// <param name="parameters">A dictionary of parameters to append to the uri.</param>
+		/// <returns>The uri with the appended parameters.</returns>
+		public static string FormatUrlParameters(string uri, Dictionary<string, string> parameters)
+		{
+			if (parameters == null || parameters.Count == 0)
+			{
+				return uri;
+			}
+
+			var stringBuilder = new StringBuilder(uri);
+
+			var firstElement = parameters.ElementAt(0);
+			stringBuilder.Append("?");

```

Comment:
You don't need to separate the first KVP from the for loop. Could just do something like

```
for (var i = 0; i < parameters.Count; i++)
{
var element = parameters.ElementAt(i);
stringBuilder.Append(i == 0 ? "?" : "&");
stringBuilder.Append(element.Key);
stringBuilder.Append("=");
stringBuilder.Append(element.Value);
}
```